src/NuGet.Core/NuGet.Frameworks/CompatibilityProvider.cs

```diff
@@ -48,11 +48,14 @@ public bool IsCompatible(NuGetFramework target, NuGetFramework candidate)
             // check the cache for a solution
             var cacheKey = new CompatibilityCacheKey(target, candidate);
 
-            bool? result = _cache.GetOrAdd(cacheKey, (Func<CompatibilityCacheKey, bool>)((key)
-                =>
-            { return IsCompatibleCore(target, candidate) == true; }));
+            bool result;
+            if (!_cache.TryGetValue(cacheKey, out result))
```

Comment:
This should work now:

```suggestion
            if (!_cache.TryGetValue(cacheKey, out bool result))
```

here it is, lol.

Reply:
Ah yes, good catch. I like that much better.